uiuc_transportation_demand/functions.py

```diff
@@ -0,0 +1,282 @@
+def units():
+
+    """
+    This function creates a dictionary linking the fuel technology
+    to its unit for clarity.
+
+    Returns:
+    --------
+    Unit: dict
+        Dictionary linking each fuel to its unit
+
+    """
+    Unit = {
+        'Gasoline': 'gallon',
+        'Diesel': 'gallon',
+        'E85': 'gallon',
+        'Hydrogen': 'kg',
+        'Electricity': 'kWh'
+    }
+
+    return Unit
+
+
+def gge_dictionary():
+
+    """
+    This function generates a hard coded dictionary of five of
+    the most relevant fleet fuel technologies and how many gallons
+    of gasoline are equivalent to one unit of each.
+
+    Returns:
+    --------
+    GGE: dict
+        The dictionary to link fuel technoligies for their GGEs
+
+    """
+
+    GGE = dict()
+
+    GGE = {
+        'Gasoline': 1.0,
+        'Diesel': 1.155,
+        'E85': 0.734,
+        'Hydrogen': 1.019,
+        'Electricity': 0.031
+    }
+
+    return GGE
+
+
+def unit_cost_dictionary():
+
+    """
+    This function generates a hard coded dictionary of the $/unit
+    for the five most relevant fuel technologies.
+
+    Returns:
+    --------
+    Cost: dict
+        Dictionary linking fuel technologies to their cost per unit ($/unit)
+
+    """
+
+    Cost = {
+        'Gasoline': 2.23,
+        'Diesel': 2.41,
+        'E85': 1.71,
+        'Hydrogen': 13.99,
+        'Electricity': 0.0426
+    }
+
+    return Cost
+
+
+def gge_cost_dictionary():
+
+    """
+    This function generates a dictionary to link each of the five
+    most relevant fuel technologies to the price of the fuel's
+    equivalent value to 1.0 gallons of gasoline.
+
+    Returns:
+    --------
+    GGECost: dict
+        A dictionary linking fuel types to their corresponding
+        price per equivalent to 1.0 gallons of gasoline.
+
+    """
+
+    GGECost = dict()
+
+    GGEDictionary = gge_dictionary()
+
+    CostDictionary = unit_cost_dictionary()
+
+    for fuel in GGEDictionary:
+        GGECost.update({fuel: CostDictionary[fuel] / GGEDictionary[fuel]})
+
+    return GGECost
+
+
+def fuel_equivalent(Fuel):
+
+    """
+    This function returns the total amount of desired fuel type
+    required to fully manage the UIUC fleet in 2019 assuming it
+    was readily available based on the fuel_data.csv
+
+    Parameters:
+    -----------
+    Fuel: string
+        Desired fuel type to analyze ('Gasoline', 'Diesel', 'E85',
+        'Hydrogen', or 'Electicity')
+
+    Returns:
+    --------
+    amount: float
+        The total amount of units for desired fuel technology that would
+        have powered the UIUC fleet in 2019
+
+    """
+
+    if type(Fuel) != str:
+        raise TypeError(
+         'Please ensure that the input parameter is of type "str".'
+        )
+
+    Dict = gge_dictionary()
+
+    if Fuel not in Dict:
+        raise IndexError(
+            f"{Fuel} not supported. Try {list(GGEDictionary.keys())}."
+        )
```

Comment:
These tests don't pass because you changed ``GGEDictionary`` to ``Dict``, I think GGEDictionary was a better name because it's more descriptive. Whatever you choose, though, they need to match (can't get keys for a dictionary you haven't created!)

Reply:
Right, I forgot I shortened the variable name to get around the PEP8 issue with character length in the following calculation lines.